src/huak/errors.rs

```diff
@@ -1,58 +1,77 @@
 use std::fmt;
 
 use anyhow::Error;
+use crate::errors::CliErrorCode::{AnyHowError, UnknownError};
+pub type CliResult<T> = std::result::Result<T, CliError>;
 
-/// Generic CLI Result type.
-pub type CliResult = Result<(), CliError>;
+trait BinaryError{}
+
+impl BinaryError for CliErrorCode {}
+impl BinaryError for Error {}
+
+#[derive(Debug)]
+pub enum CliErrorCode {
+    NotImplemented,
+    MissingVirtualEnv,
+    MissingArguments,
+    UnknownError,
+    IOError,
+    UnknownCommand,
+    DirectoryExists,
+    AnyHowError(Error)
+}
 
 #[derive(Debug)]
-/// The CLI error is the error type used at Huak's CLI-layer.
-///
-/// All errors from the lib side of Huak will get wrapped with this error.
-/// Other errors (such as command-line argument validation) will create this
-/// directly.
 pub struct CliError {
-    /// The error to display. This can be `None` in rare cases to exit with a
-    /// code without displaying a message.
-    pub error: Option<anyhow::Error>,
-    /// The process exit code.
-    pub exit_code: i32,
+    pub error: CliErrorCode
 }
 
 impl CliError {
-    /// Initialize a `CliError` with `anyhow`.
-    pub fn new(error: anyhow::Error, code: i32) -> CliError {
+    pub fn new(mut error_code: CliErrorCode) -> CliError {
         CliError {
-            error: Some(error),
-            exit_code: code,
+            error: error_code
```

Comment:
I intended for AnyhowError to simply contain any anyhow errors that appear, but just lets people use the same command. You can see this with how anyhow errors are printed

Reply:
I see. But what I mean is if I run

```
ruff .
```

and it fails it'll return with a status code right? I'd think that status code is defined by ruff or a dep ruff is using. I'd want a way to grab that status code and its stderr/debug info and pass it up through to our process when we wrap it with

```
huak lint
```

Maybe I'm just missing something, but I'm trying to understand how that case is managed.

Also on the 
```
error: error_code
```

I don't like this because it's kind of confusing. IMO Error != Error Code.

Reply:
I haven't seen any Ruff errors so far, but I could add an extra enum thats like Ruff(`ruff error type`) to contain those-
The name can totally be changed, I just used it to differentiate from the struct and enum, but they're basically the same thing. Could change to CliErrorType, perhaps?

Reply:
That's actually not a bad idea. We'd do that for ruff, black, and pytest. In the future any wrapped projects.

Reply:
I'll add them then

Reply:
I see Ruff, Black and Pytest are not part of the Rust project themselves, should I just wrap them in a String and print it out?